rust/connlib/tunnel/src/tests/sut.rs

```diff
@@ -57,8 +63,12 @@ impl StateMachineTest for TunnelTest {
     fn init_test(
         ref_state: &<Self::Reference as ReferenceStateMachine>::State,
     ) -> Self::SystemUnderTest {
+        let flux_capacitor = FluxCapacitor::default();
+
         let logger = tracing_subscriber::fmt()
             .with_test_writer()
+            // .with_writer(crate::tests::run_count_appender::appender()) // Useful for diffing logs between runs.
+            .with_timer(flux_capacitor.clone())
```

Comment:
With this custom timer, the logs of `tunnel_test` use a timestamp that is an offset in seconds from when they started. For example:

```
Running a test case with 5 transitions.
  0.000s  INFO client: snownet::node: Added new TURN server rid=5F0116F68E52DDB74F0098DCEF1CD0B2 address=Dual { v4: 203.0.113.91:3478, v6: [2001:db80::18]:3478 }
  0.000s  INFO client: snownet::node: Added new TURN server rid=990023340968434D80AAA50D65DCA19A address=Dual { v4: 203.0.113.17:3478, v6: [2001:db80::30]:3478 }
  0.000s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}: snownet::node: Added new TURN server rid=5F0116F68E52DDB74F0098DCEF1CD0B2 address=Dual { v4: 203.0.113.91:3478, v6: [2001:db80::18]:3478 }
  0.000s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}: snownet::node: Added new TURN server rid=990023340968434D80AAA50D65DCA19A address=Dual { v4: 203.0.113.17:3478, v6: [2001:db80::30]:3478 }
  0.000s DEBUG firezone_tunnel::tests::sut: No socket dst=[2001:db80::18]:3478
  0.000s DEBUG firezone_tunnel::tests::sut: No socket dst=[2001:db80::30]:3478
  0.020s  INFO relay{rid=990023340968434D80AAA50D65DCA19A}:handle_binding_request{tid=0EC0347491F0A14AF2D46143 sender=203.0.113.27:372}: firezone_relay::server: Handled BINDING request
  0.020s  INFO relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_binding_request{tid=BC6947C28EF3435F42FB6814 sender=203.0.113.27:372}: firezone_relay::server: Handled BINDING request
  0.020s  INFO relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_binding_request{tid=3473FA7A874313078B805F01 sender=[2001:db80::52]:372}: firezone_relay::server: Handled BINDING request
  0.020s  INFO relay{rid=990023340968434D80AAA50D65DCA19A}:handle_binding_request{tid=52390F873BC3BE32CAF7F2F0 sender=[2001:db80::52]:372}: firezone_relay::server: Handled BINDING request
  0.040s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.17:3478 tid=0EC0347491F0A14AF2D46143 method=binding class=success response rtt=40ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=None relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.040s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.17:3478 tid=0EC0347491F0A14AF2D46143 method=binding class=success response rtt=40ms}: snownet::allocation: Updating active socket active_socket=203.0.113.17:3478
  0.040s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=[2001:db80::30]:3478 tid=52390F873BC3BE32CAF7F2F0 method=binding class=success response rtt=40ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=Some([2001:db80::52]:372) relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.040s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=[2001:db80::30]:3478 tid=52390F873BC3BE32CAF7F2F0 method=binding class=success response rtt=40ms}: snownet::allocation: Relay supports dual-stack but we've already picked a socket active_socket=203.0.113.17:3478 additional_socket=[2001:db80::30]:3478
  0.060s DEBUG relay{rid=990023340968434D80AAA50D65DCA19A}:handle_allocate_request{tid=261AB1DB24A2B7873B78DD7E sender=203.0.113.27:372}: relay: allocate failed: Unauthorized
  0.070s  INFO relay{rid=990023340968434D80AAA50D65DCA19A}:handle_binding_request{tid=75BE615E9B02D6B080F8DC53 sender=203.0.113.76:22650}: firezone_relay::server: Handled BINDING request
  0.070s  INFO relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_binding_request{tid=6C67719801AD9D3D576C62C2 sender=203.0.113.76:22650}: firezone_relay::server: Handled BINDING request
  0.080s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.17:3478 tid=261AB1DB24A2B7873B78DD7E method=allocate class=error response rtt=40ms}: snownet::allocation: Request failed, re-authenticating error="Unauthorized"
  0.080s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.91:3478 tid=BC6947C28EF3435F42FB6814 method=binding class=success response rtt=80ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=None relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.080s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.91:3478 tid=BC6947C28EF3435F42FB6814 method=binding class=success response rtt=80ms}: snownet::allocation: Updating active socket active_socket=203.0.113.91:3478
  0.080s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=[2001:db80::18]:3478 tid=3473FA7A874313078B805F01 method=binding class=success response rtt=80ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=Some([2001:db80::52]:372) relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.080s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=[2001:db80::18]:3478 tid=3473FA7A874313078B805F01 method=binding class=success response rtt=80ms}: snownet::allocation: Relay supports dual-stack but we've already picked a socket active_socket=203.0.113.91:3478 additional_socket=[2001:db80::18]:3478
  0.090s  INFO client:handle_input{from=203.0.113.17:3478 tid=75BE615E9B02D6B080F8DC53 method=binding class=success response rtt=90ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.76:22650) srflx_ip6=None relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.090s DEBUG client:handle_input{from=203.0.113.17:3478 tid=75BE615E9B02D6B080F8DC53 method=binding class=success response rtt=90ms}: snownet::allocation: Updating active socket active_socket=203.0.113.17:3478
  0.100s  INFO relay{rid=990023340968434D80AAA50D65DCA19A}:handle_allocate_request{tid=0ADA8F88AA45385D284AD3BC sender=203.0.113.27:372 allocation=59779}: relay: Created new allocation first_relay_address=203.0.113.17 second_relay_address=2001:db80::30 lifetime=600s
  0.100s DEBUG relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_allocate_request{tid=31C4652113308C06C879EE57 sender=203.0.113.27:372}: relay: allocate failed: Unauthorized
  0.120s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.17:3478 tid=0ADA8F88AA45385D284AD3BC method=allocate class=success response rtt=40ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=Some([2001:db80::52]:372) relay_ip4=Some(203.0.113.17:59779) relay_ip6=Some([2001:db80::30]:59779) remaining_lifetime=Some(600s)
  0.130s  INFO client:handle_input{from=203.0.113.91:3478 tid=6C67719801AD9D3D576C62C2 method=binding class=success response rtt=130ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.76:22650) srflx_ip6=None relay_ip4=None relay_ip6=None remaining_lifetime=None
  0.130s DEBUG client:handle_input{from=203.0.113.91:3478 tid=6C67719801AD9D3D576C62C2 method=binding class=success response rtt=130ms}: snownet::allocation: Updating active socket active_socket=203.0.113.91:3478
  0.160s DEBUG relay{rid=990023340968434D80AAA50D65DCA19A}:handle_allocate_request{tid=F1773D9331490BCD7576B37D sender=203.0.113.76:22650}: relay: allocate failed: Unauthorized
  0.160s DEBUG gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.91:3478 tid=31C4652113308C06C879EE57 method=allocate class=error response rtt=80ms}: snownet::allocation: Request failed, re-authenticating error="Unauthorized"
  0.180s DEBUG client:handle_input{from=203.0.113.17:3478 tid=F1773D9331490BCD7576B37D method=allocate class=error response rtt=90ms}: snownet::allocation: Request failed, re-authenticating error="Unauthorized"
  0.180s  INFO relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_allocate_request{tid=82CF575105E1AE4C7BC82B1D sender=203.0.113.27:372 allocation=52200}: relay: Created new allocation first_relay_address=203.0.113.91 second_relay_address=2001:db80::18 lifetime=600s
  0.200s DEBUG relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_allocate_request{tid=46D46284A8C84FE44873F64B sender=203.0.113.76:22650}: relay: allocate failed: Unauthorized
  0.240s  INFO gateway{gid=8A0F8C1F56C1413A63BAC3C7B4EFBFBE}:handle_input{from=203.0.113.91:3478 tid=82CF575105E1AE4C7BC82B1D method=allocate class=success response rtt=80ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.27:372) srflx_ip6=Some([2001:db80::52]:372) relay_ip4=Some(203.0.113.91:52200) relay_ip6=Some([2001:db80::18]:52200) remaining_lifetime=Some(600s)
  0.250s  INFO relay{rid=990023340968434D80AAA50D65DCA19A}:handle_allocate_request{tid=4BEB7A74BD61F7A528DF86FA sender=203.0.113.76:22650 allocation=61397}: relay: Created new allocation first_relay_address=203.0.113.17 second_relay_address=2001:db80::30 lifetime=600s
  0.260s DEBUG client:handle_input{from=203.0.113.91:3478 tid=46D46284A8C84FE44873F64B method=allocate class=error response rtt=130ms}: snownet::allocation: Request failed, re-authenticating error="Unauthorized"
  0.270s  INFO client:handle_input{from=203.0.113.17:3478 tid=4BEB7A74BD61F7A528DF86FA method=allocate class=success response rtt=90ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.76:22650) srflx_ip6=None relay_ip4=Some(203.0.113.17:61397) relay_ip6=Some([2001:db80::30]:61397) remaining_lifetime=Some(600s)
  0.330s  INFO relay{rid=5F0116F68E52DDB74F0098DCEF1CD0B2}:handle_allocate_request{tid=69CFB7B7FB4531048A866762 sender=203.0.113.76:22650 allocation=60620}: relay: Created new allocation first_relay_address=203.0.113.91 second_relay_address=2001:db80::18 lifetime=600s
  0.390s  INFO client:handle_input{from=203.0.113.91:3478 tid=69CFB7B7FB4531048A866762 method=allocate class=success response rtt=130ms}: snownet::allocation: Updated allocation srflx_ip4=Some(203.0.113.76:22650) srflx_ip6=None relay_ip4=Some(203.0.113.91:60620) relay_ip6=Some([2001:db80::18]:60620) remaining_lifetime=Some(600s)
  1.090s DEBUG client:handle_timeout{active_socket=Some(203.0.113.91:3478)}: snownet::allocation: Request timed out after 1s, re-sending id=TransactionId(0xC684C970CBD6A1DA9F89025B) method=binding dst=[2001:db80::18]:3478
  1.090s DEBUG client:handle_timeout{active_socket=Some(203.0.113.17:3478)}: snownet::allocation: Request timed out after 1s, re-sending id=TransactionId(0x8125CE5398D88822014F90AB) method=binding dst=[2001:db80::30]:3478
  1.090s DEBUG firezone_tunnel::tests::sut: No socket dst=[2001:db80::18]:3478
  1.090s DEBUG firezone_tunnel::tests::sut: No socket dst=[2001:db80::30]:3478
```